src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,206 @@
+     * ResetEncoderAtBottom() command to end
+     */
+    if(m_enabled && m_isCalibrated) {
+      double pid = m_elevatorPID.calculate(getHeight());
+      double ff = m_elevatorFF.calculate(ElevatorConstants.kVelocity, ElevatorConstants.kAccel); 
+      ff = 0; 
+      double pidClamped = MathUtil.clamp(pid, -ElevatorConstants.kPowerLimit, ElevatorConstants.kPowerLimit);
+      double finalMotorPower = pidClamped + ff; 
+      
+      set(finalMotorPower);
+    } else {
+      //m_motor.feed();
+    }
+
+  }
+
+  public void close() {
+    // close the ports
+    //TODO: Is there any way to close the ports of the limit switches attached to the motor? 
+    m_topLimitSwitch.close(); 
+    m_bottomLimitSwitch.close(); 
+  }
+
+  public void set(double power) {
+    if(isBottom()){
+      if(power >0 ){ 
+        m_motor.set(power); 
+      }else{
+        m_motor.set(0); 
+      }
+    }
+
+    else if(isTop()){
+      if(power<0){
+        m_motor.set(power); 
+      }else{
+        m_motor.set(0); 
+      }
+    }
+
+    else{
+      m_motor.set(power); 
+    }
+
+  
+  }
+  
+  public void setEnabled(boolean isEnabled){
+    m_enabled = isEnabled; 
+  }
+
+  public void setIsCalibrated(){
+    m_isCalibrated = true; 
+  }
+
+
+  public void setSetpoint(double setpoint){
+    m_elevatorPID.setSetpoint(setpoint);
+  }
+
+  // public boolean getBottomLimitSwitch(){
+  //  if(m_motor.getSensorCollection().isRevLimitSwitchClosed() == 0){
+  //   return true; 
+  //  }
+  //   return false; 
+  // }
+
+  // public boolean getTopLimitSwitch(){
+  //   if(m_motor.getSensorCollection().isFwdLimitSwitchClosed() == 0){
+  //    return true; 
+  //   }
+  //    return false; 
+  // }
+
+  public boolean isBottom(){
+    if(m_bottomLimitSwitch.get() == false){
+      return true; 
+    } else{
+      return false; 
+    }
+  }
+
+  public boolean isTop(){
+    if(m_topLimitSwitch.get() == false){
+      return true; 
+    } else{
+      return false; 
+    }  }
+
+  public void resetTalonEncoder(){
+    m_talonEncoder.reset(); 
+  }
+ 
+  /**
+   * Determine the elevator height from the motor encoder.
+   * @return return height in meters
+   */
+  public double getHeight() {
+    return m_talonEncoder.getDistance(); 
+  }
+
+  /**
+   * 
+   * Zero position is the value of the absolute encoder after the elevator
+   * hits the bottom limit switch. 
+   * @return return the absolute encoder's zero position in ticks. 
+   * 
+   */
+  public boolean atSetpoint() {
+    return m_elevatorPID.atSetpoint();
+  }
+
+  public void setMotorPower(double power){
+    m_motor.set(power);
+  }
+
+  public double setMaxHeight(){
+    m_maxHeight = m_talonEncoder.getDistance();
+    return m_maxHeight;  
+  }
+
+  public void setUpElevatorTab(){
+    m_elevatorTab.addNumber("Elevator Height", ()->getHeight());
+    m_elevatorTab.add(m_elevatorPID); 
+    m_elevatorTab.addBoolean("enabled", ()->m_enabled);
+    
+  }
```

Comment:
add spaces in the lambdas to readablity and add stuff like limit switches

Reply:
added